src/hotspot/share/classfile/systemDictionaryShared.cpp

```diff
@@ -522,7 +522,7 @@ void SystemDictionaryShared::handle_class_unloading(InstanceKlass* klass) {
     remove_dumptime_info(klass);
   }
 
-  {
+  if (Arguments::is_dumping_archive() || ClassListWriter::is_enabled()) {
```

Comment:
Nit: is it worth saving the result from line 521 into a local and using it here?